internal/jobworker/cgroup/cgroup_test.go

```diff
@@ -0,0 +1,233 @@
+package cgroup
+
+import (
+	"bufio"
+	"errors"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"testing"
+)
+
+func TestCleanup(t *testing.T) {
+	if !isRoot() {
+		t.Skip("must be root to run")
+	}
+
+	dir := t.TempDir()
+	service, err := NewService(WithMountPath(dir))
+	if err != nil {
+		t.Errorf("%+v", err)
+		return
+	}
+
+	if _, err := os.Stat(service.path); err != nil {
+		t.Error(err)
+	}
+
+	expected := []string{
+		cpu,
+		io,
+		memory,
+	}
+	controllers, err := readControllers(service.path)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if !reflect.DeepEqual(controllers, expected) {
+		t.Errorf("unexpected controllers; actual: %v, expected: %v", controllers, expected)
+	}
+
+	if err := service.Cleanup(); err != nil {
+		t.Errorf("unexpected error: %+v", err)
+		return
+	}
+
+	if _, err := os.Stat(service.path); !errors.Is(err, fs.ErrNotExist) {
+		t.Errorf("expected cgroup to not exist; path: %s, error: %v", service.path, err)
+		return
+	}
+}
+
+func TestCleanupWithCgroups(t *testing.T) {
+	if !isRoot() {
+		t.Skip("must be root to run")
+	}
+
+	dir := t.TempDir()
+	service, err := NewService(WithMountPath(dir))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	if _, err := service.CreateCgroup(); err != nil {
+		t.Error(err)
+	}
+
+	if err := service.Cleanup(); err != nil {
+		t.Error(err)
+		return
+	}
+
+	if _, err := os.Stat(service.path); !errors.Is(err, fs.ErrNotExist) {
+		t.Errorf("expected cgroup to not exist; path: %s, err: %v", service.path, err)
+		return
+	}
+}
+
+func TestCleanupWithPids(t *testing.T) {
+	if !isRoot() {
+		t.Skip("must be root to run")
+	}
+
+	dir := t.TempDir()
+	service, err := NewService(WithMountPath(dir))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	cgroup, err := service.CreateCgroup()
+	if err != nil {
+		t.Error(err)
+	}
+
+	cmd := exec.Command("sleep", "30")
+	if err := cmd.Start(); err != nil {
+		t.Error(err)
+	}
+
+	if err := service.PlaceInCgroup(*cgroup, cmd.Process.Pid); err != nil {
```

Comment:
No need to spawn a command, just write a random number. We are using temp dirs after all.

Reply:
In my current tests, I'm writing to the cgroup2 filesystem, so if I don't enter a valid pid, I get back an error `no such process`. I'm going to add some new tests that do not require being root. I'll test this there.